### DISCUSSION TITLE
Filter for assigned ip addresses?

### DISCUSSION THREAD

**Discussion by slu-ymca** [netbox-community/netbox#5861] [Q&A] [answered]
Is it possible to filter for assigned ip addresses? I want to check if a specific anycast address is assigned to a specific (vm) interface. The following pynetbox / url query returns all documented ip addresses.

**pynetbox:** nb.ipam.ip_addresses.filter(assigned_object_type="virtualization.vminterface", assigned_object_id=71)
**url:** GET /api/ipam/ip-addresses/?assigned_object_type=virtualization.vm_interface&assigned_object_id=71

**Comment by DCAuto** [comment, marked as answer]
I think "device_id" or "virtual_machine_id" is what you're looking for.

 url: GET /api/ipam/ip-addresses/?virtual_machine_id=71

**Reply by slu-ymca** [reply]
This is exactly what I'm looking for. Thank you very much!

Is there any documentation about valid filters? Did you have a look at https://github.com/netbox-community/netbox/blob/develop/netbox/ipam/filters.py ?

**Reply by DCAuto** [reply]
Use the api explorer:

\<netbox-url>/api/docs/
